test/gtest/common/mem_buffer.cc

```diff
@@ -469,6 +473,11 @@ void mem_buffer::copy_between(void *dst, const void *src, size_t length,
     } else if (check_mem_types(dst_mem_type, src_mem_type, rocm_mem_types)) {
         ROCM_CALL(hipMemcpy(dst, src, length, hipMemcpyDefault));
         ROCM_CALL(hipDeviceSynchronize());
+#endif
+#if HAVE_IBV_DM
+    } else if (check_mem_types(dst_mem_type, src_mem_type, rdma_mem_types)) {
+        //TODO: implement copy using put / get operations or using memcpy with:
```

Comment:
We need that condition inside the #if HAVE_IBV_DM because otherwise rdma_mem_types won't be defined, if source or dst memory types is rdma we wouldn't want to abort the test (that way we can check ksm atomic key registration for rdma memory at minimum cost)